res/controllers/Traktor-Kontrol-S3-hid-scripts.js

```diff
@@ -0,0 +1,2227 @@
+    this.defineButton(messageShort, "!pad_6", 0x03, 0x40, 0x06, 0x80, deckFn.numberButtonHandler);
+    this.defineButton(messageShort, "!pad_7", 0x03, 0x80, 0x07, 0x01, deckFn.numberButtonHandler);
+    this.defineButton(messageShort, "!pad_8", 0x04, 0x01, 0x07, 0x02, deckFn.numberButtonHandler);
+
+    // TODO: bind touch: 0x09/0x40, 0x0A/0x02
+    this.defineButton(messageShort, "!SelectTrack", 0x0B, 0x0F, 0x0C, 0xF0, deckFn.selectTrackHandler);
+    this.defineButton(messageShort, "!LoadSelectedTrack", 0x09, 0x01, 0x09, 0x08, deckFn.loadTrackHandler);
+    this.defineButton(messageShort, "!PreviewTrack", 0x01, 0x08, 0x04, 0x10, deckFn.previewTrackHandler);
+    this.defineButton(messageShort, "!LibraryFocus", 0x01, 0x40, 0x04, 0x80, deckFn.LibraryFocusHandler);
+    this.defineButton(messageShort, "!QueueAutoDJ", 0x01, 0x20, 0x04, 0x40, deckFn.cueAutoDJHandler);
+
+    // Loop control
+    // TODO: bind touch detections: 0x0A/0x01, 0x0A/0x08
+    this.defineButton(messageShort, "!SelectLoop", 0x0C, 0x0F, 0x0D, 0xF0, deckFn.selectLoopHandler);
+    this.defineButton(messageShort, "!ActivateLoop", 0x09, 0x04, 0x09, 0x20, deckFn.activateLoopHandler);
+
+    // Rev / Flux / Grid / Jog
+    this.defineButton(messageShort, "!reverse", 0x01, 0x04, 0x04, 0x08, deckFn.reverseHandler);
+    this.defineButton(messageShort, "!slip_enabled", 0x01, 0x02, 0x04, 0x04, deckFn.fluxHandler);
+    this.defineButton(messageShort, "quantize", 0x01, 0x80, 0x05, 0x01, deckFn.quantizeHandler);
+    this.defineButton(messageShort, "!jogButton", 0x02, 0x01, 0x05, 0x02, deckFn.jogButtonHandler);
+
+    // Beatjump
+    // TODO: bind touch detections: 0x09/0x80, 0x0A/0x04
+    this.defineButton(messageShort, "!SelectBeatjump", 0x0B, 0xF0, 0x0D, 0x0F, deckFn.selectBeatjumpHandler);
+    this.defineButton(messageShort, "!ActivateBeatjump", 0x09, 0x02, 0x09, 0x10, deckFn.activateBeatjumpHandler);
+
+    // Jog wheels
+    this.defineButton(messageShort, "!jog_touch", 0x0A, 0x10, 0x0A, 0x20, deckFn.jogTouchHandler);
+    this.defineJog(messageShort, "!jog", 0x0E, 0x12, deckFn.jogHandler);
+
+    this.defineScaler(messageLong, "rate", 0x01, 0xFFFF, 0x0D, 0xFFFF, deckFn.pitchSliderHandler);
+};
+
+TraktorS3.Deck.prototype.shiftHandler = function(field) {
+    // Mixxx only knows about one shift value, but this controller has two shift buttons.
+    // This control object could get confused if both physical buttons are pushed at the same
+    // time.
+    engine.setValue("[Controls]", "touch_shift", field.value);
+    this.shiftPressed = field.value;
+    if (field.value) {
+        engine.softTakeoverIgnoreNextValue("[Master]", "gain");
+    }
+    this.controller.basicOutput(field.value, field.group, "!shift");
+};
+
+TraktorS3.Deck.prototype.playHandler = function(field) {
+    if (this.shiftPressed) {
+        engine.setValue(this.activeChannel, "start_stop", field.value);
+    } else if (field.value === 1) {
+        script.toggleControl(this.activeChannel, "play");
```

Comment:
Does this need to cancel the wheel inertia timer and/or stop scratching? I had to do that for the Kontrol S2 Mk2 mapping or it got stuck in a weird state where the deck wouldn't play after doing a backspin and quickly pressing the play button. I don't know if that's needed with the way you're directly manipulating the `scratch2` Control.

Reply:
https://github.com/mixxxdj/mixxx/blob/main/res/controllers/Traktor-Kontrol-S2-MK2-hid-scripts.js#L629

Reply:
I'll check this out